### PR TITLE
[payment-request]: crash tests - part 1

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

**Comment by marcoscaceres** [comment]
WIP, looking for initial feedback

**Comment by marcoscaceres** [comment]
@rsolomakhin, if I bump the  ABUSIVE_AMOUNT to 1,000,000, I can crash Chrome (tab).

File: payment-request/payment-request-constructor-crash.https.html [hunk 1]
```diff
@@ -0,0 +1,269 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Crash tests PaymentRequest Constructor</title>
+<link rel="help" href="https://w3c.github.io/browser-payment-api/#constructor">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+/**
```
Comment (marcoscaceres) [inline payment-request/payment-request-constructor-crash.https.html]: I'll remove this - I'm just using it as a guide while I work on the tests

**Comment by marcoscaceres** [comment]
@rsolomakhin you can test them here - all green in Canary:
https://w3c-test.org/submissions/6108/payment-request/payment-request-constructor-crash.https.html

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/237798387)

# Firefox (nightly) #
Testing web-platform-tests at revision fa2aab4eaf7996894665bd38e997836cde5fc9b1
Using browser at version BuildID 20170530100155; SourceStamp 286f71223256cbb3a769432fd860f563c4886e81
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="https://w3c-test.org/submissions/6108/payment-request/payment-request-constructor-crash.https.html">/payment-request/payment-request-constructor-crash.https.html</a></summary>


|                                         Subtest                                          | Results |                                                                                                                                                                                                           Messages                                                                                                                                                                                                          |
|------------------------------------------------------------------------------------------|---------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|                                                                                          | OK      |                                                                                                                                                                                                                                                                                                                                                                                                                             |
| `Don't crash if there is a abusive number of payment methods in the methodData sequence` | FAIL    | `assert_true: failed smoke test: @https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:81:5\nTest.prototype.step@https://web-platform.test:8443/resources/testharness.js:1406:20\ntest@https://web-platform.test:8443/resources/testharness.js:498:9\n@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:77:1\n expected true got false`   |
| `Don't crash if there is a abusive number of supported methods in one sequence`          | FAIL    | `assert_true: failed smoke test: @https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:101:5\nTest.prototype.step@https://web-platform.test:8443/resources/testharness.js:1406:20\ntest@https://web-platform.test:8443/resources/testharness.js:498:9\n@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:97:1\n expected true got false`  |
| `Don't crash if the request id has an abusive length`                                    | FAIL    | `assert_true: failed smoke test: @https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:123:5\nTest.prototype.step@https://web-platform.test:8443/resources/testharness.js:1406:20\ntest@https://web-platform.test:8443/resources/testharness.js:498:9\n@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:119:1\n expected true got false` |
| `Don't crash if PaymentDetailsInit.total.label is an abusive length`                     | FAIL    | `assert_true: failed smoke test: @https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:147:5\nTest.prototype.step@https://web-platform.test:8443/resources/testharness.js:1406:20\ntest@https://web-platform.test:8443/resources/testharness.js:498:9\n@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:143:1\n expected true got false` |
| `Don't crash if total.amount.value is an abusive length`                                 | FAIL    | `assert_true: failed smoke test: @https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:168:5\nTest.prototype.step@https://web-platform.test:8443/resources/testharness.js:1406:20\ntest@https://web-platform.test:8443/resources/testharness.js:498:9\n@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:164:1\n expected true got false` |
| `Don't crash if details.displayItems has an abusive number of items`                     | FAIL    | `assert_true: failed smoke test: @https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:191:7\nTest.prototype.step@https://web-platform.test:8443/resources/testharness.js:1406:20\ntest@https://web-platform.test:8443/resources/testharness.js:498:9\n@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:186:3\n expected true got false` |
| `Don't crash if details.shippingOptions has an abusive number of items`                  | FAIL    | `assert_true: failed smoke test: @https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:191:7\nTest.prototype.step@https://web-platform.test:8443/resources/testharness.js:1406:20\ntest@https://web-platform.test:8443/resources/testharness.js:498:9\n@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:186:3\n expected true got false` |
| `Don't crash if PaymentShippingOptions.label is an abusive length`                       | FAIL    | `assert_true: failed smoke test: @https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:214:5\nTest.prototype.step@https://web-platform.test:8443/resources/testharness.js:1406:20\ntest@https://web-platform.test:8443/resources/testharness.js:498:9\n@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:208:1\n expected true got false` |
| `Don't crash if the PaymentShippingOptions.amount.value is an abusive length`            | FAIL    | `assert_true: failed smoke test: @https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:234:5\nTest.prototype.step@https://web-platform.test:8443/resources/testharness.js:1406:20\ntest@https://web-platform.test:8443/resources/testharness.js:498:9\n@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:228:1\n expected true got false` |
| `Don't crash if PaymentItem.label is an abusive length`                                  | FAIL    | `assert_true: failed smoke test: @https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:254:5\nTest.prototype.step@https://web-platform.test:8443/resources/testharness.js:1406:20\ntest@https://web-platform.test:8443/resources/testharness.js:498:9\n@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:248:1\n expected true got false` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/237798388)

# Sauce (safari) #
Testing web-platform-tests at revision fa2aab4eaf7996894665bd38e997836cde5fc9b1
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="https://w3c-test.org/submissions/6108/payment-request/payment-request-constructor-crash.https.html">/payment-request/payment-request-constructor-crash.https.html</a></summary>


|                                         Subtest                                          | Results |                                                                                                                                                                                                        Messages                                                                                                                                                                                                        |
|------------------------------------------------------------------------------------------|---------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|                                                                                          | OK      |                                                                                                                                                                                                                                                                                                                                                                                                                        |
| `Don't crash if there is a abusive number of payment methods in the methodData sequence` | FAIL    | `assert_true: failed smoke test: https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:81:23\nstep@https://web-platform.test:8443/resources/testharness.js:1406:30\ntest@https://web-platform.test:8443/resources/testharness.js:498:22\nglobal code@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:77:5 expected true got false`   |
| `Don't crash if there is a abusive number of supported methods in one sequence`          | FAIL    | `assert_true: failed smoke test: https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:101:23\nstep@https://web-platform.test:8443/resources/testharness.js:1406:30\ntest@https://web-platform.test:8443/resources/testharness.js:498:22\nglobal code@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:97:5 expected true got false`  |
| `Don't crash if the request id has an abusive length`                                    | FAIL    | `assert_true: failed smoke test: https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:123:23\nstep@https://web-platform.test:8443/resources/testharness.js:1406:30\ntest@https://web-platform.test:8443/resources/testharness.js:498:22\nglobal code@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:119:5 expected true got false` |
| `Don't crash if PaymentDetailsInit.total.label is an abusive length`                     | FAIL    | `assert_true: failed smoke test: https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:147:23\nstep@https://web-platform.test:8443/resources/testharness.js:1406:30\ntest@https://web-platform.test:8443/resources/testharness.js:498:22\nglobal code@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:143:5 expected true got false` |
| `Don't crash if total.amount.value is an abusive length`                                 | FAIL    | `assert_true: failed smoke test: https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:168:23\nstep@https://web-platform.test:8443/resources/testharness.js:1406:30\ntest@https://web-platform.test:8443/resources/testharness.js:498:22\nglobal code@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:164:5 expected true got false` |
| `Don't crash if details.displayItems has an abusive number of items`                     | FAIL    | `assert_true: failed smoke test: https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:191:25\nstep@https://web-platform.test:8443/resources/testharness.js:1406:30\ntest@https://web-platform.test:8443/resources/testharness.js:498:22\nglobal code@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:186:7 expected true got false` |
| `Don't crash if details.shippingOptions has an abusive number of items`                  | FAIL    | `assert_true: failed smoke test: https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:191:25\nstep@https://web-platform.test:8443/resources/testharness.js:1406:30\ntest@https://web-platform.test:8443/resources/testharness.js:498:22\nglobal code@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:186:7 expected true got false` |
| `Don't crash if PaymentShippingOptions.label is an abusive length`                       | FAIL    | `assert_true: failed smoke test: https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:214:23\nstep@https://web-platform.test:8443/resources/testharness.js:1406:30\ntest@https://web-platform.test:8443/resources/testharness.js:498:22\nglobal code@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:208:5 expected true got false` |
| `Don't crash if the PaymentShippingOptions.amount.value is an abusive length`            | FAIL    | `assert_true: failed smoke test: https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:234:23\nstep@https://web-platform.test:8443/resources/testharness.js:1406:30\ntest@https://web-platform.test:8443/resources/testharness.js:498:22\nglobal code@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:228:5 expected true got false` |
| `Don't crash if PaymentItem.label is an abusive length`                                  | FAIL    | `assert_true: failed smoke test: https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:254:23\nstep@https://web-platform.test:8443/resources/testharness.js:1406:30\ntest@https://web-platform.test:8443/resources/testharness.js:498:22\nglobal code@https://web-platform.test:8443/payment-request/payment-request-constructor-crash.https.html:248:5 expected true got false` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/237798398)

# Chrome (unstable) #
Testing web-platform-tests at revision 397649ebe5e00ae4168c6682e261d1a652594b52
Using browser at version 60.0.3112.7 dev
Starting 10 test iterations

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/237798399)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 397649ebe5e00ae4168c6682e261d1a652594b52

**Comment by marcoscaceres** [comment]
Happy to report Firefox doesn't crash running these :)

**Review by sideshowbarker — APPROVED** [review]

**Comment by TakayoshiKochi** [comment]
Hi there - Chromium team imported this test in their tree and now it is marked flaky
(to be specific, "Pass Timeout" in the expectation).
context: https://bugs.chromium.org/p/chromium/issues/detail?id=732839

I'd like to ask a couple of questions about the expectation of this test:
1. the filename suggests it expects browser to crash, but for browser developers this test should run without crash, right?
2. how fast/slow is this test expected to run (if successful)?  Can it take, say, 10 seconds, even when successful?

For 2, if the test is expected to finish fast (<1sec, it looks so when I run it locally on my Linux box with Chromium) timeout is something we should worry about.

**Comment by marcoscaceres** [comment]
We can definitely scale it down a bit more. We want these tests to be aggressive in trying to crash the browser, but not so much as to become a problem.

**Comment by TakayoshiKochi** [comment]
Do you mean reducing `ABUSIVE_AMOUNT` down to smaller value?

Maybe I'll try `<meta name="timeout" content="long">` first and see if it fixes.

**Pull request merged**